### PR TITLE
Reset state and set enable form actions for add role sidebar.

### PR DESCRIPTION
**Pull request by SantoshPisini**
## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

Resolves #20839 

## What is the new behavior?

Using the form action buttons are not disabled & the value is being reset.

## Additional context

`hasChanges={true}` is set to `true` as to enable form action buttons, and if the user clicks on 'Save' with an empty value, form validation is triggered, hence I feel this is valid as the user will be blocked until they enter a valid value.

[create_role_sidebar_fix.webm](https://github.com/supabase/supabase/assets/39261271/c42fd77f-3bd2-43e8-9ac3-e236a7bfe4dd)

### CODE REVIEW

**Comment by SantoshPisini** [comment]
@saltcod Please review :)

**Comment by SantoshPisini** [comment]
@encima Is there any tag that I need to add for this PR to review?

**Comment by encima** [comment]
nope, you have done the hard work, we just need to review it!

File: apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx [hunk 1]
```diff
@@ -87,8 +87,8 @@ const CreateRolePanel = ({ visible, onClose }: CreateRolePanelProps) => {
           <FormActions
             form={formId}
             isSubmitting={isCreating}
-            hasChanges={form.formState.isDirty}
-            handleReset={() => form.reset()}
+            hasChanges={true}
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx:90]: quick question - this doesnt look right - it should still be `form.formState.isDirty` right?

Comment (SantoshPisini) [reply apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx:90]: `hasChanges={true}` is set to `true ` as to enable form action buttons, and if the user clicks on 'Save' with an empty value, form validation is triggered, hence I feel this is valid as the user will be blocked until they enter a valid value.

**Review by joshenlim — APPROVED** [review]

File: apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ const CreateRolePanel = ({ visible, onClose }: CreateRolePanelProps) => {
 
   const handleClose = () => {
     onClose()
-    form.reset()
+    form.reset(initialValues)
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx:74]: this isn't necessary i think

```suggestion
    form.reset()
```

Comment (joshenlim) [reply apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx]: sorry! was wrong here - reverting the change

**Pull request merged**